diagrams/metametamodel-gen.puml

```diff
@@ -15,32 +11,31 @@ class ConceptInterface extends FeaturesContainer
 
 class Containment extends Link
 
-abstract class Datatype extends MetamodelElement
+abstract class DataType extends MetamodelElement
+
+class Enumeration extends DataType implements NamespaceProvider
 
 abstract class Feature extends NamespacedEntity {
-  multiplicity: Multiplicity
+  optional: boolean
   derived: boolean
 }
 
 abstract class FeaturesContainer extends MetamodelElement implements NamespaceProvider {
   allFeatures(): List<Feature>
 }
 
-abstract class Link extends Feature
+' primitive type: "JSON"
```

Comment:
This comment is not clear to me

Reply:
It's just a result of the primitive types being defined in the self-definition of the M3/lioncore. Because PlantUML doesn't have a 1st-class construct for primitive types I just emit them as comments.